plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -30,12 +30,13 @@ module.exports = function DeleteSiteActionModule(pb) {
         var self = this;
         var siteid = self.pathVars.siteid;
         var siteQueryService = new pb.SiteQueryService();
+        var SiteService = new pb.SiteService();
         siteQueryService.getCollections(function(err, allCollections) {
-            siteQueryService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
-                if(util.isError(err)) {
+            SiteService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
```

Comment:
Yes, it would.  To solve this problem, inside of the SiteService, I would register a function for the AFTER_DELETE that would:
* retrieve the collection list
* iterate over collections deleting documents that contain the given site id.

SiteService.js
```
SiteService.prototype.deleteSiteSpecificContent = function(siteUid, options, cb) {

  //retrieve list of collections
  //for each collection
    //delete from collection[i] where siteId = 'xyz'
 //callback when done
};

SiteService.afterDelete = function(context, cb) {
  var siteObj = context.data;
  context.service.deleteSiteSpecificContent(siteObj.uid, cb);
};

//Event Registries
BaseObjectService.on(TYPE + '.' + BaseObjectService.AFTER_DELETE, SiteService.afterDelete);
```

This is basically a reversal of the logic in the implementation you have for ```deleteSiteSpecificContent```.  The approach I am suggesting will simplify the amount of code you have to write.  You will not need to verify the site exists because the delete single check does that for you and you don't worry about manually having to delete the site object after all other site data was deleted.  All you will need in the outlined ```deleteSiteSpecificContent``` function is what is on 584-597 of site_service.js and the code to retrieve the collections list.